### DISCUSSION TITLE
Kwikset HC620

### DISCUSSION THREAD

**Discussion by Shakey001** [zwave-js/zwave-js-ui#3124] [General]
Included a HC620 Kwikset lock, works fine but cannot add user codes. Programed codes directly into the lock locally they work, zwavejsui doesn't pull them from lock either. Older 910's work great.
Gives an error- "Driver: The user code must consist of 4 to 10 of the following characters: 0123456789" 

Tried adding 1234 as code

Gives a "Value update failed' erro popup.

Tried excluding and re-including.
Using Zooz 800usb stick, latest zwavejsui build.

**Comment by robertsLando** [comment]
Please make a [driver log](https://zwave-js.github.io/zwave-js-ui/#/troubleshooting/generating-logs?id=driver-logs), loglevel `debug` and attach it here as a file (drag & drop into the text field).

**Reply by Shakey001** [reply]
Let me know if you need
 anything else.

Thank you

[zwave-js-ui-store.zip](https://github.com/zwave-js/zwave-js-ui/files/11671256/zwave-js-ui-store.zip)

**Reply by robertsLando** [reply]
@AlCalzone any clue why this happens? I also noticed this error in other locks after latest releases 2/3

**Reply by AlCalzone** [reply]
> zwavejsui doesn't pull them from lock either

You have to refresh them once manually. This doesn't happen automatically, since querying user codes has been found to generate a ton of traffic.

That said, I don't know exactly what's going wrong because we obfuscate user codes in logs. It seems like the lock first says "nope, can't do that"
```
2023-06-06T18:02:13.793Z DRIVER » [Node 020] [REQ] [SendDataBridge]
                                  │ source node id:   1
                                  │ transmit options: 0x25
                                  │ callback id:      9
                                  └─[Security2CCMessageEncapsulation]
                                    │ sequence number: 230
                                    │ extensions:
                                    │ · type: SPAN
                                    │   sender EI: 0x1d78be0335199931213c16638f6f02a9
                                    └─[SupervisionCCGet]
                                      │ session id:      8
                                      │ request updates: true
                                      └─[UserCodeCCExtendedUserCodeSet]
                                          code #1: **** (status: Enabled)

2023-06-06T18:02:13.843Z DRIVER « [Node 020] [REQ] [BridgeApplicationCommand]
                                  │ RSSI: -90 dBm
                                  └─[Security2CCMessageEncapsulation]
                                    │ sequence number: 202
                                    └─[SupervisionCCReport]
                                        session id:          8
                                        more updates follow: false
                                        status:              Fail
                                        duration:            0s
```

and then sends us an update with the user code (or is it the old one?)
```
2023-06-06T18:02:13.908Z DRIVER « [Node 020] [REQ] [BridgeApplicationCommand]
                                  │ RSSI: -90 dBm
                                  └─[Security2CCMessageEncapsulation]
                                    │ sequence number: 203
                                    └─[UserCodeCCExtendedUserCodeReport]
                                        code #1:      **** (status: Enabled)
                                        next user id: 0
```

One thing we can try is to use the normal `UserCodeSet` command. Using `ExtendedUserCodeSet` feels unintended. Normally this shouldn't make a difference, but... it's Z-Wave, you never know.

**Reply by AlCalzone** [reply]
@Shakey001 how are you running Z-Wave JS UI? Are you able to run a test build?
https://zwave-js.github.io/zwave-js-ui/#/development/custom-docker?id=building-a-container-using-dockerfilecontrib

**Reply by Shakey001** [reply]
Running it on RPi4 on docker.
Container attached

Thanks for the help.
[Zwavejsui_container.txt](https://github.com/zwave-js/zwave-js-ui/files/11678851/Zwavejsui_container.txt)

**Reply by robertsLando** [reply]
@AlCalzone The easier thing is you make a test build and I create a PR with that and make a test docker release with it

**Reply by Shakey001** [reply]
Sounds good just let me know how to help.

**Comment by robertsLando** [comment]
@Shakey001 I started a workflow that will build a test image, you can test it using `test` tag in your docker-compose file instead of `latest`

Check the status of the build here: https://github.com/zwave-js/zwave-js-ui/actions/runs/5210870905

(should take 1h from now)

**Reply by Shakey001** [reply]
Thanks, installed a little bit ago, can now set and modify user codes now will play with over next couple days. Let me know if you need anything.
Very close to finishing up a driver so zwave-js-ui can be used with Remote Technologies Inc. systems.

**Reply by robertsLando** [reply]
Ok so this fixes your issue so. @AlCalzone I think you should merge that

**Reply by AlCalzone** [reply]
PR merged

**Reply by robertsLando** [reply]
Released?

**Reply by AlCalzone** [reply]
In 10.23.3, bit more than 2 weeks ago :D

**Reply by robertsLando** [reply]
Ok so it's available with 8.19 :)

**Reply by Shakey001** [reply]
Thank you both again for help.

**Reply by Shakey001** [reply]
Might have spoke to soon, after changing to latest the lock no locker responds to lock or unlock command.

**Reply by robertsLando** [reply]
@Shakey001 driver logs? May be something else now

**Reply by robertsLando** [reply]
Try to use this image: `docker pull zwavejs/zwave-js-ui:sha-4721f91`

It's the exact commit before the v11, if that works it means it's something related to v11

**Reply by Shakey001** [reply]
Have it working on latest. It was acting weird, excluded, re-included all seems good.  Thanks
